### DISCUSSION TITLE
Big O of createEntity adapter

### DISCUSSION THREAD

**Discussion by Sivaranjith1** [reduxjs/redux-toolkit#3389] [Q&A]
Does the createEntity adapter have a O(1) or does is scale with the amount of data. We are experience some issues where having more data would cause the `dispatch(entity.actions.updateMany())` to be slower.

**Comment by phryneas** [comment]
It should be in `O(numberOfUpdatedItems)` plus the overhead of `immer`.

How many items are you talking about here?

**Reply by Sivaranjith1** [reply]
Its a 1000 elements, and I update 4 of them very frequently. Some are updated on mouse move

**Reply by phryneas** [reply]
That would not take a lot of time, but "on mouse move" could be hundreds of times per second. You need to debounce that to something more sensible like one dispatch per second or so.

**Reply by Sivaranjith1** [reply]
Debouncing sounds like a good idea. I however tested with `console.time` for the `updateMany` from the create entityAdapter. It however does not look like `O(1) + overhead` since in both cases I only update when property

With a lot of elements in the state
![image](https://user-images.githubusercontent.com/24842210/233563670-63354345-bd6a-468f-9acb-7701536ce6c2.png)

With few elements in the state
![image](https://user-images.githubusercontent.com/24842210/233563703-3287451c-fe48-4c14-9c38-6fc2a1e114cf.png)

![image](https://user-images.githubusercontent.com/24842210/233563801-af954251-6a7a-4ce4-847b-9c803444dd44.png)

**Reply by phryneas** [reply]
Going by https://immerjs.github.io/immer/performance/, you would see those 50ms probably in an environment when changing 5k of 50k, not when changing only 4 of 1000.

If you look at the [source code of updateMany](https://github.com/reduxjs/redux-toolkit/blob/98d7dda57992ccd20b93eaf85bcf35ac802b0b3f/packages/toolkit/src/entities/unsorted_state_adapter.ts#L129), you can see that what happens here on our side is essentially -  it's not a lot of code, and everything there scales by the number of updates, not with the length of your array.

**Comment by markerikson** [comment]
What entity adapter methods are you using?

(also, this should be in the Redux Toolkit repo - moving it over there)
